internal/engine/expression.go

```diff
@@ -17,14 +17,14 @@ func (e Engine) evaluateExpression(ctx ExecutionContext, expr command.Expr) (typ
 	}
 
 	switch ex := expr.(type) {
+	case command.BinaryExpression:
+		return e.evaluateBinaryExpr(ctx, ex)
 	case command.ConstantBooleanExpr:
```

Comment:
One's `Expression` and others are `Expr` is there a reason? Shouldn't we aim for consistency?